### PR TITLE
ICU-22723 RC API change reports for C/J

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22723
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/APIChangeReport.md [hunk 1]
```diff
@@ -23,11 +23,49 @@ Removed from ICU 75
   
 | File | API | ICU 75 | ICU 76 |
 |---|---|---|---|
+| messageformat2_data_model.h | Builder&amp; icu::message2::MFDataModel::Builder::addUnsupportedStatement(UnsupportedStatement&amp;&amp;, UErrorCode&amp;) |  InternalICU 75 | (missing)
```
Comment (markusicu) [inline icu4c/APIChangeReport.md:26]: @catamorphism @mihnita @echeran There are a lot of MF2 changes here, and below for Java -- both compared with what the API change report showed just a couple of weeks ago. Are these all covered by API proposals?

**Review by markusicu — APPROVED** [review]

File: icu4c/APIChangeReport.md (hunk 1 above)
Comment (catamorphism) [reply icu4c/APIChangeReport.md:26]: Yes, this is due to https://github.com/unicode-org/icu/pull/3177 (Java) and https://github.com/unicode-org/icu/pull/3193 (C++). This removes unsupported expressions/statements from MF2, which includes removing some methods/classes from the data model. I followed Mihai's lead in not submitting a design doc. Should I have submitted one?

Comment (markusicu) [reply icu4c/APIChangeReport.md:26]: API changes should always be proposed via icu-design email and a section in the API proposal doc.
Even for tech preview, it doesn't make sense to publicly propose the initial API and then silently change parts of it.

Comment (catamorphism) [reply icu4c/APIChangeReport.md:26]: Got it.

**Comment by markusicu** [comment]
Thanks Mihai for checking!
Ok with me on no proposals for the DataModel.
It's probably useful to include a short section about the tech preview changes and status in the download page, but not a blocker for the release candidate.
@richgillam FYI

**Pull request merged**